File: pset2/vigenere/vigenere.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int getShift (char c)
{
    if (isupper(c))
    {
        return c - 65;
    }
    else
    {
        return c - 97;
    }
}

char shiftChar(int shift, char c)
{
    if (isupper(c))
    {
        if (c + shift <= 90)
        {
            return c + shift;
        }
        else
        {
            return c + shift - 90 + 65 - 1;
        }
    }
    else
    {
        if (c + shift <= 122)
        {
            return c + shift;
        }
        else
        {
            return c + shift - 122 + 97 - 1;
        }
    }
}

int main(int argc, string argv[])
{

    string plaintext, key;
    int i, j;
    i = 0;
    j = 0;
    key = argv[argc - 1];
    if (argc != 2)
    {
        return 1;
    }
    for (int k = 0; k < strlen(key); k++)
    {
        if (!isalpha(key[k]))
        {
            return 1;
        }
    }
    printf("plaintext: ");
    plaintext = get_string();
    printf("ciphertext: ");
    while (i < strlen(plaintext))
    {
        if (isalpha(plaintext[i]))
        {
            int shift = getShift(key[j % strlen(key)]);
            printf("%c", shiftChar(shift, plaintext[i]));
            i++;
            j++;
        }
        else
        {
            if (plaintext[i] == ' ')
            {
                printf(" ");
            }
            else
            {
                printf("%c", plaintext[i]);
            }
            i++;
        }
    }
    printf("\n");

    return 0;
}

File: pset6/sentimental/vigenere/vigenere.py
import sys


def getShift(c):
    if(c.isupper()):
        return ord(c) - 65
    else:
        return ord(c) - 97


def shiftChar(shift, c):
    if(c.isupper()):
        if(ord(c) + shift <= 90):
            return chr(ord(c) + shift)
        else:
            return chr(ord(c) + shift - 90 + 65 - 1)

    else:
        if(ord(c) + shift <= 122):
            return chr(ord(c) + shift)
        else:
            return chr(ord(c) + shift - 122 + 97 - 1)


def main(argc, argv):
    if(argc != 2):
        exit(1)
    key = argv[argc - 1]
    for i in range(0, len(key)):
        if(not key[i].isalpha()):
            exit(1)
    plaintext = input("plaintext: ")
    print("ciphertext: ", end="")
    j = 0
    for i in range(0, len(plaintext)):
        if(plaintext[j].isalpha()):
            shift = getShift(key[j % len(key)])
            print(shiftChar(shift, plaintext[i]), end="")
            j += 1
        else:
            if(plaintext[i] == " "):
                print(" ", end="")
            else:
                print(plaintext[i], end="")
    print("")


main(len(sys.argv), sys.argv)
